src/main/java/mate/academy/DAO/BookDaoImpl.java

```diff
@@ -0,0 +1,103 @@
+package mate.academy.DAO;
+
+import mate.academy.model.Book;
+import mate.academy.exception.DataProcessingException;
+
+import mate.academy.lib.Dao;
+import mate.academy.util.ConnectionUtil;
+
+import java.math.BigDecimal;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String INSERT_BOOK_QUERY = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL_QUERY = "SELECT * FROM books";
+    private static final String UPDATE_BOOK_QUERY = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BY_ID_QUERY = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(INSERT_BOOK_QUERY, Statement.RETURN_GENERATED_KEYS)) {
```

Comment:
Consider using a separate private method to handle the logic of setting the `id` on the `book` object to avoid duplicating this logic in different methods.